python/ray/serve/_private/publish_provider.py

```diff
@@ -49,14 +49,15 @@ class AnyscalePublishProvider(PublishProvider):
     def publish(
         config: Dict,
         *,
-        name: str,
+        name: Optional[str],
         ray_version: str,
         base_image: Optional[str] = None,
     ):
         service_config = {
-            "name": name,
             "ray_serve_config": config,
         }
+        if name is not None:
+            service_config["name"] = name
```

Comment:
An Anyscale service should require the "name" field right - so if the user doesn't specify a name, the publish command will just error out? Should we handle this more explicitly somehow?

Reply:
Ah this is the point of the change :) in workspaces we'll default the name to the workspace name if it's not provided